subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CrossVersionIntegrationSpec.groovy

```diff
@@ -31,6 +32,7 @@ import static spock.lang.Retry.Mode.SETUP_FEATURE_CLEANUP
 
 @CrossVersionTest
 @Retry(condition = { RetryConditions.onIssueWithReleasedGradleVersion(instance, failure) }, mode = SETUP_FEATURE_CLEANUP, count = 2)
+@CleanupTestDirectory
```

Comment:
was this not being cleaned up before?

Reply:
The question is, why has been cleaned up before even when the test failed (that is the effect of the annotation).

I don't know, any ideas? It seems like an oversight.

Reply:
@big-guy 
> was this not being cleaned up before?

As I understand it, without this annotation the integration test will **unconditionally** clean up the test directory. However, we don't want to do this in case the test fails, and this is what adding this annotation achieves.